### PR TITLE
Ftr 4172 matriculation status

### LINKED ISSUE #4172: Abi-statuksen näyttäminen opiskelijalle
**Opinnot** päätoiminnon **ylioppilaskijoitukseni** välilehdellä näytetään opiskelijan abi-status seuraavasti:

-Mikäli opiskelijalla on abi-status tämän kertovan tekstin ohella näytetään opiskelijalle painike **Ilmoittaudu YO-kokeerseen** josta opiskelija päätyy sähköiseen ylioppilaskokeen ilmoittautumisjärjestelmään.
-Mikäli opiskelijalla ei ole abi-statusta näytetään hänelle teksti jossa kerrotaan mitä vaaditaan abi-statuksen saavuttamiseksi.
- Mikäli opiskelijalla on abi-status ja on jo ilmoittautunut yo-kokeeseen hänelle näytetään ilmoittautumispäivämäärä sekä milloin yo-kokeet ovat.

### PR DESCRIPTION
**Pull request by ilmoeuro**
Fixes #4172 matriculation status in studies view

### CODE REVIEW

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/current-record.tsx [hunk 1]
```diff
@@ -65,6 +65,7 @@ class CurrentRecord extends React.Component<CurrentRecordProps, CurrentRecordSta
       <div className="application-sub-panel__header application-sub-panel__header--studies-detailed-info" key={this.props.records.current.workspace.id}>
         {this.props.records.current.workspace.name} {this.props.records.current.workspace.nameExtension && "(" + this.props.records.current.workspace.nameExtension + ")"}
       </div>
+      <div>MAKKKARAAAA!</div>
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/current-record.tsx]: I think this does not belong here?

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/yo.tsx [hunk 2]
```diff
@@ -1,64 +1,68 @@
...
-    return this.props.i18n.text.get(`plugin.records.hops.matriculationSubject.${code}`);
-  }
 
-  /**
-   * Component did mount life-cycle method  
-   * 
-   * Reads available matriculation subjects from REST API
-   */
   componentDidMount() {
+    mApi().records.studentMatriculationEligibility
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/yo.tsx]: Can we use async and '~/util/promisify'?

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/current-record.tsx (hunk 1 above)
Comment (ilmoeuro) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/current-record.tsx]: removed

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/yo.tsx (hunk 2 above)
Comment (ilmoeuro) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/yo.tsx]: Added async and promisify

**Pull request merged**